qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationElwinTabView.cpp

```diff
@@ -169,7 +181,77 @@ void InelasticDataManipulationElwinTabView::setup() {
   setHorizontalHeaders();
 }
 
-IndirectPlotOptionsView *InelasticDataManipulationElwinTabView::getPlotOptions() { return m_uiForm.ipoPlotOptions; }
+IndirectPlotOptionsView *InelasticDataManipulationElwinTabView::getPlotOptions() const {
+  return m_uiForm.ipoPlotOptions;
+}
+
+void InelasticDataManipulationElwinTabView::subscribePresenter(IElwinPresenter *presenter) { m_presenter = presenter; }
+
+void InelasticDataManipulationElwinTabView::notifyRunClicked() { m_presenter->handleRunClicked(); }
+
+void InelasticDataManipulationElwinTabView::notifySaveClicked() { m_presenter->handleSaveClicked(); }
+
+void InelasticDataManipulationElwinTabView::notifyPlotPreviewClicked() { m_presenter->handlePlotPreviewClicked(); }
+
+void InelasticDataManipulationElwinTabView::notifyFilesFound() { m_presenter->handleFilesFound(); }
+
+void InelasticDataManipulationElwinTabView::notifySelectedSpectrumChanged(int index) {
+  m_presenter->handlePreviewSpectrumChanged(index);
+}
+
+void InelasticDataManipulationElwinTabView::notifyPreviewIndexChanged(int index) {
+  m_presenter->handlePreviewIndexChanged(index);
+}
+
+void InelasticDataManipulationElwinTabView::notifyRemoveDataClicked() { m_presenter->handleRemoveSelectedData(); }
+
+void InelasticDataManipulationElwinTabView::notifyAddWorkspaceDialog() { showAddWorkspaceDialog(); }
+
+std::unique_ptr<IAddWorkspaceDialog>
+InelasticDataManipulationElwinTabView::getAddWorkspaceDialog(QWidget *parent) const {
+  return std::make_unique<IndirectAddWorkspaceDialog>(parent);
+}
+
+void InelasticDataManipulationElwinTabView::showAddWorkspaceDialog() {
+  if (!m_addWorkspaceDialog)
+    m_addWorkspaceDialog = getAddWorkspaceDialog(this->parentWidget());
+  m_addWorkspaceDialog->setWSSuffices(getSampleWSSuffices());
```

Comment:
Can you add this line?
```
m_addWorkspaceDialog->setAttribute(Qt::WA_DeleteOnClose);
```
It makes sure the Dialog object is deleted when it is closed. I'm not sure if this happens at the moment